plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3217,6 +3217,22 @@ public void testProjectionPushdownOnPartitionedTables()
         }
     }
 
+    @Test
+    public void testProjectionPushdownOnPartitionedTablesWithComments()
+    {
+        try {
+            assertUpdate("CREATE TABLE table_with_partition_comments (id BIGINT COMMENT 'id', qid BIGINT COMMENT 'QID', root ROW(f1 BIGINT, f2 BIGINT) COMMENT 'root') WITH (partitioning = ARRAY['id'])");
+            assertUpdate("INSERT INTO table_with_partition_comments VALUES (1, 1, ROW(1, 2)), (1, 2, ROW(2, 3)), (1, 3, ROW(3, 4))", 3);
+            assertQuery("SELECT id, root.f2 FROM table_with_partition_comments", "VALUES (1, 2), (1, 3), (1, 4)");
+            assertQuery("SELECT id, root.f2 FROM table_with_partition_comments where id = 1 and qid = 1 and root.f1 = 1", "VALUES (1, 2)");
+            assertQuery("SELECT id, root.f2 FROM table_with_partition_comments where qid = 2 and root.f1 = 2", "VALUES (1, 3)");
+            assertQuery("SELECT id, root.f2 FROM table_with_partition_comments where id = 1 and qid = 1", "VALUES (1, 2)");
```

Comment:
What's the difference between these test queries?
please add  a comment for each of them explaining what case it's covering